### ISSUE TITLE
prometheus-operator creashes when using OAuth2 without setting `tlsConfig`

### ISSUE THREAD

**Issue by kne-zyres** [prometheus-operator/prometheus-operator#6946]
### Is there an existing issue for this?

- [X] I have searched the existing issues

## Description

The operator crashes when reconciling a `ServiceMonitor` CRD that is using OAuth2 with a nil tlsConfig. According to the [documentation](https://prometheus-operator.dev/docs/api-reference/api/#monitoring.coreos.com/v1.OAuth2) this field is optional.

## Steps to Reproduce

Create a ServiceMonitor. (or any other resource with a `OAuth2` sub-resource)

```
apiVersion: monitoring.coreos.com/v1
kind: ServiceMonitor
metadata:
  name: test
spec:
  endpoints:
  - honorLabels: true
    oauth2:
      # tlsConfig: null
      clientId:
        secret:
          key: client-id
          name: prometheus-client
      clientSecret:
        key: client-secret
        name: prometheus-client
      tokenUrl: http://test/oauth/token
    path: /api/prometheus-metrics/general
    port: default
```

## Expected Result

The resource is reconciled.

## Actual Result

prometheus-operator crashes. This can be worked around by providing an empty value for tlsConfig.

### Prometheus Operator Version

```shell
v0.76.1
```

### Kubernetes Version

```yaml
v1.29.5+k3s1
```

### Kubernetes Cluster Type

Other (please comment)

### How did you deploy Prometheus-Operator?

helm chart:prometheus-community/kube-prometheus-stack

### prometheus-operator log output

```shell
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1855ee2]

goroutine 385 [running]:
github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1.(*SafeTLSConfig).Validate(0x0)
	/home/runner/work/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1/types.go:713 +0x22
github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1.(*OAuth2).Validate(0x1?)
	/home/runner/work/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1/types.go:610 +0x10c
github.com/prometheus-operator/prometheus-operator/pkg/assets.(*StoreBuilder).AddOAuth2(0xc000ed1d80, {0x29c3c90, 0xc00057eaa0}, {0xc000f84d50, 0x18}, 0xc000a18870)
	/home/runner/work/prometheus-operator/prometheus-operator/pkg/assets/store.go:127 +0x73
github.com/prometheus-operator/prometheus-operator/pkg/prometheus.(*ResourceSelector).SelectServiceMonitors(0xc000bc6140, {0x29c3c90, 0xc00057eaa0}, 0xc000a61c48)
	/home/runner/work/prometheus-operator/prometheus-operator/pkg/prometheus/resource_selector.go:170 +0x1467
github.com/prometheus-operator/prometheus-operator/pkg/prometheus/server.(*Operator).createOrUpdateConfigurationSecret(0xc000626c88, {0x29c3c90, 0xc00057eaa0}, 0xc00109e008, 0xc000a186c0, {0xc00098fb90, 0x1, 0x1}, 0xc000ed1d80)
	/home/runner/work/prometheus-operator/prometheus-operator/pkg/prometheus/server/operator.go:1138 +0x12d
github.com/prometheus-operator/prometheus-operator/pkg/prometheus/server.(*Operator).sync(0xc000626c88, {0x29c3c90, 0xc00057eaa0}, {0xc000895780, 0x32})
	/home/runner/work/prometheus-operator/prometheus-operator/pkg/prometheus/server/operator.go:821 +0x6ac
github.com/prometheus-operator/prometheus-operator/pkg/prometheus/server.(*Operator).Sync(0xc000626c88, {0x29c3c90?, 0xc00057eaa0?}, {0xc000895780, 0x32})
	/home/runner/work/prometheus-operator/prometheus-operator/pkg/prometheus/server/operator.go:767 +0x2c
github.com/prometheus-operator/prometheus-operator/pkg/operator.(*ResourceReconciler).processNextReconcileItem(0xc000568100, {0x29c3c90, 0xc00057eaa0})
	/home/runner/work/prometheus-operator/prometheus-operator/pkg/operator/resource_reconciler.go:443 +0x231
github.com/prometheus-operator/prometheus-operator/pkg/operator.(*ResourceReconciler).Run.func1()
	/home/runner/work/prometheus-operator/prometheus-operator/pkg/operator/resource_reconciler.go:406 +0x3f
golang.org/x/sync/errgroup.(*Group).Go.func1()
	/home/runner/go/pkg/mod/golang.org/x/sync@v0.8.0/errgroup/errgroup.go:78 +0x56
created by golang.org/x/sync/errgroup.(*Group).Go in goroutine 247
	/home/runner/go/pkg/mod/golang.org/x/sync@v0.8.0/errgroup/errgroup.go:75 +0x96
```

**Comment by slashpai** [comment]
This is fixed in `v0.76.2` https://github.com/prometheus-operator/prometheus-operator/releases/tag/v0.76.2

**Comment by kne-zyres** [comment]
Splendid! Thank you.
